src/platform/modulesmanager/tests/ModuleManagerTests/src/ModulesManagerTests.cpp

```diff
@@ -85,6 +102,117 @@ namespace Tests
         ASSERT_EQ(MMI_OK, mm->MpiSet(componentName1.c_str(), "", payload, ARRAY_SIZE(payload)));
     }
 
+    TEST_F(ModuleManagerTests, MpiSetDesiredSingleComponent)
+    {
+        constexpr const char payload[] = R""""(
+            [
+                {
+                    "TestComponent1": {
+                        "testParameter": "testValue"
+                    }
+                }
+            ])"""";
+
+        ASSERT_EQ(MPI_OK, mm->MpiSetDesired(clientName.c_str(), payload, ARRAY_SIZE(payload)));
+    }
+
+    TEST_F(ModuleManagerTests, MpiSetDesiredMultipleComponents)
+    {
+        constexpr const char payload[] = R""""(
+            [
+                {
+                    "TestComponent1": {
+                        "testParameter": "testValue"
+                    },
+                    "TestComponent2": {
+                        "testParameter": "testValue"
+                    }
+                }
+            ])"""";
+
+        ASSERT_EQ(MPI_OK, mm->MpiSetDesired(clientName.c_str(), payload, ARRAY_SIZE(payload)));
+    }
+
+    TEST_F(ModuleManagerTests, MpiSetDesiredMultipleConfigurations)
+    {
+        constexpr const char payload[] = R""""(
+            [
+                {
+                    "TestComponent1": {
+                        "testParameter": "testValue"
+                    }
+                },
+                {
+                    "TestComponent1": {
+                        "testParameter": "testValue"
+                    },
+                    "TestComponent2": {
+                        "testParameter": "testValue"
+                    }
+                }
```

Comment:
This is not sufficient

We need: 

- Components with 0, 1 and N objects
- Objects with 0, 1 and N settings
- Settings with either missing, wrong or good type, invalid and valid values

Reply:
Updated tests to clarify the use of objects vs settings.